### PR TITLE
Add Vault 0.7 and remove old versions that should not be run any longer

### PR DESCRIPTION
**Pull request by jefferai**

### CODE REVIEW

**Comment by yosifkit** [comment]
```diff
diff --git a/_bashbrew-list b/_bashbrew-list
index 34e436b..73cbdde 100644
--- a/_bashbrew-list
+++ b/_bashbrew-list
@@ -1,6 +1,4 @@
-vault:0.6
-vault:0.6.2
-vault:0.6.3
 vault:0.6.4
 vault:0.6.5
+vault:0.7.0
 vault:latest
diff --git a/vault_0.6.2/Dockerfile b/vault_0.6.2/Dockerfile
deleted file mode 100644
index 549cbfb..0000000
diff --git a/vault_0.6.2/docker-entrypoint.sh b/vault_0.6.2/docker-entrypoint.sh
deleted file mode 100755
index afd8d1a..0000000
diff --git a/vault_0.6.3/Dockerfile b/vault_0.6.3/Dockerfile
deleted file mode 100644
index ec858e6..0000000
diff --git a/vault_0.6.3/docker-entrypoint.sh b/vault_0.6.3/docker-entrypoint.sh
deleted file mode 100755
index 1f55e1a..0000000
diff --git a/vault_latest/Dockerfile b/vault_0.6.5/Dockerfile
similarity index 100%
copy from vault_latest/Dockerfile
copy to vault_0.6.5/Dockerfile
diff --git a/vault_0.6.4/docker-entrypoint.sh b/vault_0.6.5/docker-entrypoint.sh
similarity index 100%
copy from vault_0.6.4/docker-entrypoint.sh
copy to vault_0.6.5/docker-entrypoint.sh
diff --git a/vault_latest/Dockerfile b/vault_latest/Dockerfile
index e3ccce7..8cb0336 100644
--- a/vault_latest/Dockerfile
+++ b/vault_latest/Dockerfile
@@ -2,7 +2,7 @@ FROM alpine:3.4
 MAINTAINER Jeff Mitchell <jeff@hashicorp.com> (@jefferai)
 
 # This is the release of Vault to pull in.
-ENV VAULT_VERSION=0.6.5
+ENV VAULT_VERSION=0.7.0
 
 # This is the release of https://github.com/hashicorp/docker-base to pull in order
 # to provide HashiCorp-built versions of basic utilities like dumb-init and gosu.
@@ -14,7 +14,7 @@ RUN addgroup vault && \
     adduser -S -G vault vault
 
 # Set up certificates, our base tools, and Vault.
-RUN apk add --no-cache ca-certificates gnupg openssl libcap && \
+RUN apk add --no-cache ca-certificates gnupg openssl libcap jq && \
     gpg --recv-keys 91A6E7F85D05C65630BEF18951852D87348FFC4C && \
     mkdir -p /tmp/build && \
     cd /tmp/build && \
diff --git a/vault_latest/docker-entrypoint.sh b/vault_latest/docker-entrypoint.sh
index ae7f397..451f1b4 100755
--- a/vault_latest/docker-entrypoint.sh
+++ b/vault_latest/docker-entrypoint.sh
@@ -12,10 +12,12 @@ set -e
 # VAULT_*_ADDR is also set, the resulting URI will combine the protocol and port
 # number with the IP of the named interface.
 get_addr () {
-    IF_NAME=$1
-    ip addr show dev $IF_NAME | awk -v uri=$2 '/\s*inet\s/ { \
+    local if_name=$1
+    local uri_template=$2
+    ip addr show dev $if_name | awk -v uri=$uri_template '/\s*inet\s/ { \
       ip=gensub(/(.+)\/.+/, "\\1", "g", $2); \
-      print gensub(/^(.+:\/\/).+(:.+)$/, "\\1" ip "\\2", "g", uri)}'
+      print gensub(/^(.+:\/\/).+(:.+)$/, "\\1" ip "\\2", "g", uri); \
+      exit}'
 }
 
 if [ -n "$VAULT_REDIRECT_INTERFACE" ]; then
```

File: library/vault [hunk 1]
```diff
@@ -1,8 +1,6 @@
 # maintainer: Jeff Mitchell <jeff@hashicorp.com> (@jefferai)
 
-0.6: git://github.com/hashicorp/docker-vault@c087a27b5ec93445e89dc46f25af973f114f1399 0.6.5
```
Comment (yosifkit) [inline library/vault]: Did you want to leave this as an alias for the newest `0.6.x` release (maybe add one for `0.7`)?  Up to you. :smile:

Comment (jefferai) [inline library/vault]: Yeah, I thought about it but the thing is, 0.X doesn't *mean* much in terms of compatibility...it's not (at least currently) like e.g. postgres where migrating to a different version means running upgrades are required. So from my perspective, I'd rather leave a few specific older releases around for those that are running it but need access to that same container, and then encourage people to simply use the latest version.

**Comment by yosifkit** [comment]
Build test of #2791; 302ca47 (`vault`):

```console

$ bashbrew build vault:0.6.4
Building bashbrew/cache:3278a11c650641c73830a2dafa7bed386d793c45fc29cc4e9f0ebe267fce4933 (vault:0.6.4)
Tagging vault:0.6.4

$ test/run.sh vault:0.6.4
testing vault:0.6.4
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build vault:0.6.5
Building bashbrew/cache:fa16d19809e19a39dbf8be3c82e01982c8e000de4fc14492e124e73e5c58e91b (vault:0.6.5)
Tagging vault:0.6.5

$ test/run.sh vault:0.6.5
testing vault:0.6.5
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed


$ bashbrew build vault:0.7.0
Building bashbrew/cache:02cd9ef6b7030123b0adc7b93c8ebcf503f03c0720f69ba6b2276dfd020a88ae (vault:0.7.0)
Tagging vault:0.7.0
Tagging vault:latest

$ test/run.sh vault:0.7.0
testing vault:0.7.0
	'utc' [1/4]...passed
	'cve-2014--shellshock' [2/4]...passed
	'no-hard-coded-passwords' [3/4]...passed
	'override-cmd' [4/4]...passed

```

**Pull request merged**